frontend/server/src/Controllers/ProblemBookmark.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace OmegaUp\Controllers;
+
+/**
+ * ProblemBookmarkController
+ *
+ * @psalm-type ProblemListItem=array{accepted: int, alias: string, can_be_removed?: bool, difficulty: float|null, difficulty_histogram: list<int>, points: float, problem_id: int, quality: float|null, quality_histogram: list<int>, quality_seal: bool, ratio: float, score: float, submissions: int, tags: list<array{name: string, source: string}>, title: string, visibility: int}
+ * @psalm-type BookmarkCreateResponse=array{success: bool}
+ * @psalm-type BookmarkDeleteResponse=array{success: bool}
+ * @psalm-type BookmarkExistsResponse=array{bookmarked: bool}
+ * @psalm-type BookmarkListResponse=array{problems: list<ProblemListItem>, total: int}
+ */
+class ProblemBookmark extends \OmegaUp\Controllers\Controller {
+    /**
+     * Bookmark a problem for the current user
+     *
+     * @omegaup-request-param string $problem_alias
+     *
+     * @return array{success: bool}
+     */
+    public static function apiCreate(\OmegaUp\Request $r): array {
+        $r->ensureIdentity();
+        $problemAlias = $r->ensureString(
+            'problem_alias',
+            fn (string $alias) => \OmegaUp\Validators::alias($alias)
+        );
+
+        // Validate that the problem exists
+        $targetProblem = \OmegaUp\DAO\Problems::getByAlias($problemAlias);
+        if (is_null($targetProblem) || is_null($targetProblem->problem_id)) {
+            throw new \OmegaUp\Exceptions\NotFoundException('problemNotFound');
+        }
+
+        // Check if already bookmarked
+        $currentUserId = $r->identity->identity_id;
+        if (
+            \OmegaUp\DAO\ProblemBookmarks::isProblemBookmarked(
+                $currentUserId,
+                $targetProblem->problem_id
+            )
+        ) {
+            // Already bookmarked, return success
+            return ['success' => true];
```

Comment:
For consistency with the other endpoints, can you use here:
```suggestion
            return ['status' => 'ok'];
```
?